### PR TITLE
[dy] Read DB credentials from AWS secret manager

### PR DESCRIPTION
**Pull request by dy46**
# Description

Add a way to populate the `db_connection_url` from AWS secret manager. The environment variable is called `AWS_DB_SECRETS_NAME`, and expects the name of the AWS secret manager secret. I will update the documentation once this PR is finalized.

# How Has This Been Tested?

- [x] Tested locally with AWS secret manager

# Checklist
- [x] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [x] I have performed a self-review of my own code
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

cc: @wangxiaoyou1993

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
Was this PR tested?

File: mage_ai/orchestration/db/setup.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        except Exception as e:
+            print("Unable to fetch secrets from AWS Secrets Manager", e)
+    elif os.getenv(PG_DB_USER):
+        db_user = os.getenv(PG_DB_USER)
+        db_pass = os.getenv(PG_DB_PASS)
+        db_name = os.getenv(PG_DB_NAME)
+        db_host = os.getenv(PG_DB_HOST, '127.0.0.1')
+        db_port = os.getenv(PG_DB_PORT, '5432')
+
+    if db_user:
```
Comment (wangxiaoyou1993) [inline mage_ai/orchestration/db/setup.py:43]: should we require all the five config values?

File: mage_ai/orchestration/db/setup.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    if os.getenv(AWS_DB_SECRETS_NAME):
+        try:
+            response = get_secret(os.getenv(AWS_DB_SECRETS_NAME))
+            secrets = json.loads(response)
+
+            if secrets and secrets.get('engine') == 'postgres':
+                db_user = secrets.get('username')
+                db_pass = secrets.get('password')
+                db_name = secrets.get('dbname')
+                db_host = secrets.get('host', '127.0.0.1')
```
Comment (wangxiaoyou1993) [inline mage_ai/orchestration/db/setup.py:31]: default host and port can be constants

**Comment by dy46** [comment]
@wangxiaoyou1993 updated

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**